python/lsst/afw/coord/refraction.py

```diff
@@ -0,0 +1,284 @@
+    """
+    temperature = _extractTemperature(weather, useKelvin=True)
+    waterVaporPressure = humidityToPressure(weather)
+    airPressure = _extractPressure(weather)
+    dryPressure = airPressure - waterVaporPressure
+
+    eqn = (dryPressure/cds.mbar)*(57.90E-8 - 9.3250E-4*u.Kelvin/temperature +
+                                  0.25844*u.Kelvin**2/temperature**2.)
+
+    densityFactor = float((1. + eqn)*(dryPressure/cds.mbar)/(temperature/u.Kelvin))
+
+    return densityFactor
+
+
+def densityFactorWater(weather):
+    """Calculate water vapor pressure term to refractive index calculation.
+
+    This replicates equation 16 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        Returns the relative density of water vapor at the given pressure and temperature.
+    """
+    temperature = _extractTemperature(weather, useKelvin=True)
+    waterVaporPressure = humidityToPressure(weather)
+
+    densityEqn1 = float(-2.37321E-3 + 2.23366*u.Kelvin/temperature -
+                        710.792*u.Kelvin**2/temperature**2. +
+                        7.75141E-4*u.Kelvin**3/temperature**3.)
+
+    densityEqn2 = float(waterVaporPressure/cds.mbar)*(1. + 3.7E-4*waterVaporPressure/cds.mbar)
+
+    relativeDensity = float(waterVaporPressure*u.Kelvin/(temperature*cds.mbar))
+    densityFactor = (1 + densityEqn2*densityEqn1)*relativeDensity
+
+    return densityFactor
+
+
+def humidityToPressure(weather):
+    """Simple function that converts humidity and temperature to water vapor pressure.
+
+    This replicates equations 18 & 20 of Stone 1996 "An Accurate Method for Computing Atmospheric Refraction"
+
+    Parameters
+    ----------
+    weather : lsst.afw.coord Weather
+        Class containing the measured temperature, pressure, and humidity
+        at the observatory during an observation
+
+    Returns
+    -------
+    float
+        The water vapor pressure in millibar, calculated from the given humidity and temperature.
+    """
+    if np.isnan(weather.getHumidity()):
+        humidity = lsstWeather.getHumidity()
+    else:
+        humidity = weather.getHumidity()
+    x = np.log(humidity/100.0)
+    temperature = _extractTemperature(weather)
+    temperatureEqn1 = (temperature + 238.3*u.Celsius)*x + 17.2694*temperature
+    temperatureEqn2 = (temperature + 238.3*u.Celsius)*(17.2694 - x) - 17.2694*temperature
+    dewPoint = 238.3*float(temperatureEqn1/temperatureEqn2)
+
+    waterVaporPressure = (4.50874 + 0.341724*dewPoint + 0.0106778*dewPoint**2 + 0.184889E-3*dewPoint**3 +
+                          0.238294E-5*dewPoint**4 + 0.203447E-7*dewPoint**5)*133.32239*u.pascal
+
+    return waterVaporPressure
+
+
+def _extractTemperature(weather, useKelvin=False):
```

Comment:
Why does this function get a leading underscore when the others don't? You're declaring, "you should never have to use this", but if it's useful why wouldn't you want it to be used? There are no implementation-specific details in here. You've already made this less-than-public by not including it in `__all__`.